src/map/script.c

```diff
@@ -3439,43 +3439,53 @@ static int set_reg(struct script_state *st, struct map_session_data *sd, int64 n
 	if (is_string_variable(name)) {// string variable
 		const char *str = (const char*)value;
 
+		if (strlen(str) > SCRIPT_STRING_VAR_LENGTH) {
```

Comment:
How about only limiting the variables that are persisted to the database? Their use in scripts is (or, should) be far less common than the non-persisted variables, and unless there's a compelling reason to limit the maximum string length I'd avoid it at all costs.

Reply:
Sure, easy to do... But I thought you're a friend of having consistency. That's why I made no exception for temporary variables.

Reply:
I agree with the consistency part, but it's also true that most use cases for string variables only care about scope variables, and are never saved to the database.

I'm mostly thinking of long menu sequences constructed by concatenating options into a string (which the client limits to 127 options if I remember correctly, and we surely want each of them to be able to be more than 1 character long, not including the `:`), or of long text to print through mes (I believe the client can support up to 4k characters in a single message dialog, with some NEMO patches applied?), so it makes sense not to add an artificial restriction on those strings unless it's necessary for technical reasons